src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs

```diff
@@ -328,159 +328,174 @@ private int RunCore(TextWriter consoleOutput, CancellationToken cancellationToke
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            var analyzerOptions = new AnalyzerOptions(ImmutableArray.Create<AdditionalText, AdditionalTextFile>(additionalTextFiles));
-
-            Func<ImmutableArray<Diagnostic>> getAnalyzerDiagnostics = null;
-            if (!analyzers.IsDefaultOrEmpty)
-            {
-                var analyzerManager = new AnalyzerManager();
-                var analyzerExceptionDiagnostics = new ConcurrentSet<Diagnostic>();
-                Action<Diagnostic> addExceptionDiagnostic = diagnostic => analyzerExceptionDiagnostics.Add(diagnostic);
-
-                var analyzerDriver = AnalyzerDriver.Create(compilation, analyzers, analyzerOptions, analyzerManager, addExceptionDiagnostic, out compilation, cancellationToken);
-                getAnalyzerDiagnostics = () =>
-                    {
-                        var analyzerDiagnostics = analyzerDriver.GetDiagnosticsAsync().Result;
-                        return analyzerDiagnostics.AddRange(analyzerExceptionDiagnostics);
-                    };
-            }
-
-            // Print the diagnostics produced during the parsing stage and exit if there were any errors.
-            if (PrintErrors(compilation.GetParseDiagnostics(), consoleOutput))
-            {
-                return Failed;
-            }
-
-            if (PrintErrors(compilation.GetDeclarationDiagnostics(), consoleOutput))
+            CancellationTokenSource analyzerCts = null;
+            try
```

Comment:
I agree.  It's gotten a bit out of hand and needs to be broken up.  Will take a shot at it.